charts/metering-ansible-operator/templates/crds/meteringconfig.crd.yaml

```diff
@@ -367,55 +367,146 @@ spec:
                       type: object
                       properties:
+                        enabled:
+                          type: boolean
                         authenticatedEmails:
                           type: object
                           properties:
+                            enabled:
+                              type: boolean
                             createSecret:
                               type: boolean
                             data:
                               type: string
-                            enabled:
-                              type: boolean
+                              minLength: 1
                             secretName:
                               type: string
+                              minLength: 1
+                          oneOf:
+                          - required:
+                            - enabled
+                            properties:
+                              enabled:
+                                enum:
+                                - false
+                            allOf:
+                            - not:
+                                required:
+                                - createSecret
+                            - not:
+                                required:
+                                - data
+                            - not:
+                                required:
+                                - secretName
+                          - required:
+                            - enabled
+                            - createSecret
+                            - secretName
+                            properties:
+                              enabled:
+                                enum:
+                                - true
+                              createSecret:
+                                enum:
+                                - false
+                            allOf:
+                            - not:
+                                required:
+                                - seed
+                          - required:
+                            - enabled
+                            - createSecret
+                            - data
+                            properties:
+                              enabled:
+                                enum:
+                                - true
+                              createSecret:
+                                enum:
+                                - true
                         cookie:
                           type: object
                           properties:
                             createSecret:
                               type: boolean
                             secretName:
                               type: string
+                              minLength: 1
                             seed:
                               type: string
+                              minLength: 32
+                          oneOf:
```

Comment:
This would enforce the following configurations (but no way to enforce that `tls.enabled: false` is set:
```
reporting-operator:
  spec:
    authProxy:
      cookie:
        createSecret: true
        seed: non-empty-seed
        # secretName is optional
```
or:
```
reporting-operator:
  spec:
    authProxy:
      cookie:
        createSecret: false
        secretName: existing-secret-name
```

Reply:
In the case where `spec.tls.enabled` is omitted, or uses the default value, `true`, then these values would be overridden by the ansible-operator.